flink-core/src/main/java/org/apache/flink/api/common/operators/base/ReduceOperatorBase.java

```diff
@@ -54,6 +54,32 @@
 @Internal
 public class ReduceOperatorBase<T, FT extends ReduceFunction<T>> extends SingleInputOperator<T, T, FT> {
 
+	/**
+	 * An enumeration of hints, optionally usable to tell the system exactly how to execute the combiner phase
+	 * of a reduce.
+	 * (Note: The final reduce phase (after combining) is currently always executed by a sort-based strategy.)
+	 */
+	public enum CombineHint {
```

Comment:
Should we consider an additional flag {{CombineHint.NONE}} for cases where the expected number of duplicate keys is relatively small (i.e. FLINK-3279)?